fast64_internal/oot/oot_utility.py

```diff
@@ -379,6 +380,35 @@ def getCustomProperty(data, prop):
 	value = getattr(data, prop)
 	return value if value != "Custom" else getattr(data, prop + str("Custom"))
 
+def getActorExportValue(detailedProp, idField, field, entranceProp):
+	if idField == 'Custom' or (entranceProp is not None and entranceProp.customActor):
+		field = getCustomPropName(field)
+		return getattr(detailedProp, field, None)
+	else:
+		if field != 'actorID' and field != 'transActorID':
+			if field == 'actorParam':
+				return getActorParameter(detailedProp, detailedProp.actorKey, 'Params', field)
+			elif field == 'transActorParam':
+				return getActorParameter(detailedProp, detailedProp.transActorKey, 'Params', field)
+			else:
+				dpKey = detailedProp.actorKey
+				actorType = getattr(detailedProp, dpKey + '.type', None)
+				for actorNode in root:
+					if dpKey == actorNode.get('Key'):
+						for elem in actorNode:
+							target = elem.get('Target')
+							if hasActorTiedParams(elem.get('TiedParam'), actorType):
+								if target == field == 'XRot':
+									return getActorParameter(detailedProp, dpKey, 'XRot', None)
+								elif target == field == 'YRot':
+									return getActorParameter(detailedProp, dpKey, 'YRot', None)
+								elif target == field == 'ZRot':
+									return getActorParameter(detailedProp, dpKey, 'ZRot', None)
```

Comment:
afaict this is equivalent:
```suggestion
			elif field in {"XRot", "YRot", "ZRot"}:
				dpKey = detailedProp.actorKey
				actorType = getattr(detailedProp, dpKey + '.type', None)
				for actorNode in root:
					if dpKey == actorNode.get('Key'):
						for elem in actorNode:
							target = elem.get('Target')
							if hasActorTiedParams(elem.get('TiedParam'), actorType):
								if target == field:
									return getActorParameter(detailedProp, dpKey, field, None)
								
									
								
									
```
which makes me wonder why `getActorExportValue` is a single function since it splits most of its behavior by an argument, so it should in theory be better split into several functions?

Reply:
I did the split, it makes more sense that way